File: benchmarks/tsb/bench_read_table.ts
/**
 * Benchmark: readTable — parse a 100k-row tab-separated string
 */
import { readTable } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 2;
const ITERATIONS = 5;

// Build TSV string (tab-separated)
const lines = ["id\tvalue\tlabel"];
for (let i = 0; i < ROWS; i++) {
  lines.push(`${i}\t${(i * 1.1).toFixed(4)}\tcat_${i % 50}`);
}
const tsvContent = lines.join("\n");

for (let i = 0; i < WARMUP; i++) {
  readTable(tsvContent);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  readTable(tsvContent);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "read_table",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_read_table.py
"""Benchmark: read_table — parse a 100k-row tab-separated file"""
import json, time
import pandas as pd

ROWS = 100_000
WARMUP = 2
ITERATIONS = 5

# Build TSV file
tmp_path = "/tmp/gh-aw/agent/bench_read_table.tsv"
with open(tmp_path, "w") as f:
    f.write("id\tvalue\tlabel\n")
    for i in range(ROWS):
        f.write(f"{i}\t{i * 1.1:.4f}\tcat_{i % 50}\n")

for _ in range(WARMUP):
    pd.read_table(tmp_path)

start = time.perf_counter()
for _ in range(ITERATIONS):
    pd.read_table(tmp_path)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "read_table",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
